api wrapper (#2)
grepr/grepr-tools: api-wrapper/README.md

# Grepr Query API wrapper

> [Grepr](http://grepr.ai/) **Intelligent Observability Data Engine** is a proxy for obserability agents. It collects the data from agents such as:

Reviewer: ```suggestion
[Grepr](http://grepr.ai/) **Intelligent Observability Data Engine** is a proxy for observability agents. It collects the data from agents such as:
```

> Datadog, Splunk, New Relic and continuously semantically analyses the data stream. It automatically detetcts similarities in the

Reviewer: ```suggestion
Datadog, Splunk, New Relic and continuously semantically analyses the data stream. It automatically detects similarities in the
```

> data and using machine learns builds and maintains an active set of filters. Frequently occuring data is summarised and forwarded

Reviewer: ```suggestion
data and using machine learning builds and maintains an active set of filters. Frequently occurring data is summarised and forwarded
```

> to the observability backend while unique data is passed straigh through. This results in a data reduction of 90% with the

Reviewer: ```suggestion
to the observability backend while unique data is passed straight through. This results in a data reduction of 90% with the
```

associated platform cost savings.

No data is lost. All data intercepted by Grepr is retained in low cost storage and can be queried and optionally backfilled to
the observability platform. This can be performed via the Grepr dashboard or via the API. This is where this set of utilities
come in handy.

## Installation

Written for Python3

Create a virtual environment for Python. In the directory where you downloaded these files.

```bash
$ python3 -m venv venv
```

> Active the virtual environment

Reviewer: ```suggestion
Activate the virtual environment
```

```bash

> $ . venv/bin/active

Reviewer: ```suggestion
$ . venv/bin/activate
```

```

Install the dependencies

```bash

> $ pip install -r requirments.txt

Reviewer: ```suggestion
$ pip install -r requirements.txt
```
